src/avro_json_decoder.erl

```diff
@@ -266,68 +275,70 @@ type_from_name(_)                 -> undefined.
 %%% Values parsing
 %%%===================================================================
 
-parse_value(null, Type, _ExtractFun) when ?AVRO_IS_NULL_TYPE(Type) ->
-  avro_primitive:null();
+parse_value(Value, Type, ExtractFun) ->
+  parse(Value, Type, ExtractFun, _IsWrapped = true).
 
-parse_value(V, Type, _ExtractFun) when ?AVRO_IS_BOOLEAN_TYPE(Type) andalso
-                                       is_boolean(V) ->
+parse(Value, TypeName, ExtractFun, IsWrapped) when is_list(TypeName) ->
+  %% Type is defined by its name
+  Type = ExtractFun(TypeName),
+  parse(Value, Type, ExtractFun, IsWrapped);
+parse(Value, Type, _ExtractFun, IsWrapped) when ?AVRO_IS_PRIMITIVE_TYPE(Type) ->
+  WrappedValue = parse_prim(Value, Type),
```

Comment:
No, not adding clauses. What I mean is everything is the same except parse_prim clauses return 
e.g. {fun avro_primitive:int/1, V}, and here you call WrapFun(V) when IsWrapped =:= true.

Minor thing, agreed.